src/IO/Archives/hasRegisteredArchiveFileExtension.cpp

```diff
@@ -6,7 +6,9 @@ namespace DB
 
 bool hasRegisteredArchiveFileExtension(const String & path)
 {
-    return path.ends_with(".zip") || path.ends_with(".zipx");
+    return path.ends_with(".zip") || path.ends_with(".zipx") || path.ends_with(".tar") || path.ends_with(".tar.gz")
```

Comment:
please put all the extensions in `static constexpr std::array` and use `std::any_of`